homeassistant/components/openweathermap/config_flow.py

```diff
@@ -45,7 +45,9 @@ async def async_step_user(self, user_input=None):
             latitude = user_input[CONF_LATITUDE]
             longitude = user_input[CONF_LONGITUDE]
 
-            await self.async_set_unique_id(f"{latitude}-{longitude}")
+            await self.async_set_unique_id(
+                f"{latitude}-{longitude}-{user_input[CONF_MODE]}"
+            )
```

Comment:
Honestly, I don't think we should set a unique ID as all for this integration.
lat/long is just a matter of a small tick change and voila, a new unique ID.

I suggest we remove it.

Reply:
I agree on the principle. 
Problem is that entities without unique_ids are quite second-class citizens, currently, and that the "This entity does not have a unique ID" message in the UI might/will be considered as a regression by users.

Reply:
> Problem is that entities without unique_ids are quite second-class citizens, currently, and that the "This entity does not have a unique ID" message in the UI might/will be considered as a regression by users.

In this case, the entry_id can be used of the configuration entry as an alternative.